src/http/conn.rs

```diff
@@ -610,6 +617,8 @@ impl<K: Key, T: Transport, H: MessageHandler<T>> Conn<K, T, H> {
             return ReadyResult::Continue(self);
         }
 
+        events = events | EventSet::hup();
```

Comment:
epoll docs suggest that `EPOLLHUP` is always listened for. May not be the case in kqueue/windows, I suppose...